docs: add dot prod & scalar_select (#2242)
zama-ai/tfhe-rs: tfhe/docs/fhe-computation/operations/dot-product.md

# Dot Product

This document details the dot product operations supported by **TFHE-rs**.

| name          | symbol         | type   |
|---------------|----------------| ------ |
| Dot Product   | `dot_product`  | Binary |

> The dot product currently only supports the case were one of the operand is a slice 
> of `FheBool` and the other is a slice of clear values (e.g. `u64`).
> Both slices must have the same size.

Reviewer: ```suggestion
Currently, the dot product supports the following case:
- One operand is a slice of `FheBool`
- The other operand is a slice of clear values (e.g., `u64`)
- Both slices must be of the same length
```

> The following example shows how to perform dot_product:

Reviewer: ```suggestion
The following example shows how to perform a dot product:
```
